internal/store/bundle.go

```diff
@@ -175,6 +238,70 @@ func (b *bundleStore) storePath(ref reference.Reference) (string, error) {
 	return storeDir, nil
 }
 
+// scanAllBundles scans the bundle store directories and creates the internal map of App image
+// references. This function must be called before any other public BundleStore interface method.
+func (b *bundleStore) scanAllBundles() error {
+	if err := filepath.Walk(b.path, b.processBundleStoreFile); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (b *bundleStore) processBundleStoreFile(path string, info os.FileInfo, err error) error {
+	idRefPath := filepath.Join(b.path, "_ids")
+	if err != nil {
+		return err
+	}
+
+	if info.IsDir() {
+		return nil
+	}
+
+	if !strings.HasSuffix(info.Name(), ".json") {
+		return nil
+	}
+
+	if strings.HasPrefix(path, idRefPath) {
+		rel := path[len(idRefPath)+1:]
+		dg := strings.Split(filepath.ToSlash(rel), "/")[0]
+		id := ID{digest.NewDigestFromEncoded(digest.SHA256, dg)}
+		appendRef(b.refsMap, id, id)
+		return nil
+	}
+
+	ref, err := b.pathToReference(path)
+	if err != nil {
+		return err
+	}
+	bndl, err := b.fetchBundleJSON(path)
+	if err != nil {
+		return err
+	}
+	id, err := FromBundle(bndl)
+	if err != nil {
+		return err
+	}
+	if _, ok := b.refsMap[id]; !ok {
+		b.refsMap[id] = []reference.Reference{ref}
+	} else {
+		b.refsMap[id] = append(b.refsMap[id], ref)
```

Comment:
I think `append` handles a nil slice, so you can remove the `if` and use directly `b.refsMap[id] = append(b.refsMap[id], ref)`